xla/service/gpu/gpu_latency_hiding_scheduler.cc

```diff
@@ -237,12 +237,27 @@ bool GpuScheduleCrossesOverlapLimit(
 
       // If candidate can be overlapped with in-flight collectives
       bool can_overlap = true;
-      for (const auto occupier :
+      for (const auto async_occupier :
            sched_state.resource_occupiers_in_flight.at(resource_type)) {
-        if (sched_state.async_tracker->IsSupportedAsyncStart(*occupier)) {
+        if (sched_state.async_tracker->IsSupportedAsyncStart(*async_occupier)) {
+          HloInstruction* occupier =
+              async_occupier->opcode() == HloOpcode::kAsyncStart
+                  ? async_occupier->async_wrapped_instruction()
+                  : const_cast<HloInstruction*>(async_occupier);
```

Comment:
Const cast is discouraged because it makes the code less safe.

Why do you need it? `GetAsyncReplicaGroups` accepts const.